packages/actions/src/Mine/mineHandlerStateRoot.spec.ts

```diff
@@ -0,0 +1,251 @@
+import { createAddress } from '@tevm/address'
+import { type TevmNode, createTevmNode } from '@tevm/node'
+import { parseEther } from '@tevm/utils'
+import { describe, expect, it } from 'vitest'
+import { callHandler } from '../Call/callHandler.js'
+import { mineHandler } from '../Mine/mineHandler.js'
+import { setAccountHandler } from '../SetAccount/setAccountHandler.js'
+
+// These issues happens both in `debugTraceBlockJsonRpcProcedure` and `debugTraceTransactionJsonRpcProcedure`,
+// but are globally applicable.
+//
+// First issue:
+// What happens is that the state root is not available for the parent block, which is the one we just mined.
+// This happens only in very specific conditions. Each of these tests will:
+// 1. Run one or multiple transactions, with various parameters that will trigger the issue or not
+// 2. Mine the block
+// 3. Get the parent block, and check if its state root is available
+// -> a. works when we specify origin or from, no matter how many txs we ran in the parent block
+// -> b. fails if we don't specify origin or from AND run 2 txs or more
+// -> c. ... but the latter will work if it's only one tx ran from the default (non-specified) account
+// -> d. fails if the balance of the caller was set too high? threshold here is pretty inconsistent
+//
+// Second issue:
+// What happens is that mining the block itself will fail with a balance error.
+// Each of these tests will:
+// 1. Run one or multiple transactions, with various parameters that will trigger the issue or not
+// 2. Attempt to mine the block
+// -> a. fails whatever the parameters are if we specify skipBalance: true
+// -> b. fails if the balance of the caller was set to 0 (which is probably the expected behavior)
+// -> c. but works if we set the balance of the caller to just 1 or more
+// -> ... which is weird, because 1 would not be enough, and at some point if it becomes too high
+// -> ... it will later trigger the first issue (see at 3.d)
+describe('mineHandler parent state root issue', () => {
+	describe('Retrieving the parent state root', () => {
+		const mineAndTestParentStateRoot = async (client: TevmNode) => {
+			await mineHandler(client)({ blockCount: 1 })
+
+			const vm = await client.getVm()
+			const block = await vm.blockchain.getCanonicalHeadBlock()
+			const parentBlock = await vm.blockchain.getBlock(block.header.parentHash)
+			const hasParentStateRoot = await vm.stateManager.hasStateRoot(parentBlock.header.stateRoot)
+			expect(hasParentStateRoot).toBe(true)
```

Comment:
Is this the bug? 
Did you try setting `loggingLevel: "debug"`?

This is the implementation for hasStateRoot:
```typescript
import { bytesToHex } from '@tevm/utils'

/**
 * Returns true if state root exists
 * @type {import("../state-types/index.js").StateAction<'hasStateRoot'>}
 */
export const hasStateRoot = (baseState) => (root) => {
	return Promise.resolve(baseState.stateRoots.has(bytesToHex(root)))
}
```

Have you tried console.logging vm.stateManager._baseState.stateRoots?

Finally have you tried looking to see where mineHandler sets the state root and console logging there?